projects/ion/src/lib/notification/notification.component.ts

```diff
@@ -0,0 +1,79 @@
+import { Component, ElementRef, Input, ViewChild } from '@angular/core';
+import { StatusType } from '../core/types';
+import { IconType } from '../icon/icon.component';
+
+export interface NotificationProps {
+  title: string;
+  message: string;
+  type?: StatusType;
+  icon?: IconType;
+  fixed?: boolean;
+}
+
+@Component({
+  selector: 'ion-notification',
+  templateUrl: './notification.component.html',
+  styleUrls: ['./notification.component.scss'],
+})
+export class NotificationComponent {
+  @Input() title!: NotificationProps['title'];
+  @Input() message!: NotificationProps['message'];
+  @Input() icon?: NotificationProps['icon'];
+  @Input() type?: NotificationProps['type'] = 'success';
+  @Input() fixed?: NotificationProps['fixed'] = false;
+  @ViewChild('notificationRef', { static: false }) notification: ElementRef;
+
+  private timer: ReturnType<typeof setTimeout>;
+
+  public getIcon(): IconType {
+    const icons = {
+      success: 'check-solid',
+      info: 'info-solid',
+      warning: 'exclamation-solid',
+      negative: 'close-solid',
+    };
+    return icons[this.type];
+  }
+
+  public getClass() {
+    if (this.icon) {
+      return 'default-icon';
+    }
+    return `default-icon ${this.type}-icon`;
```

Comment:
```suggestion
    return `default-icon ${ this.icon || this.type + '-icon' }`;
```
I didn't understand this logic, but I think this is a simplest way to do it.